engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/EjeaBackendIsCopyingCachedOutputsSpec.scala

```diff
@@ -0,0 +1,131 @@
+    )
+
+    CallCachingModes.filter(_.readFromCache) foreach { mode =>
+      initialHashDataAndEjhaResponseCombinations filter { _.validForMode(mode) } foreach { case combo @ InitialHashDataAndEjhaResponseCombination(hashComboName, initialHashData, ejhaResponse, finalHashData, _) =>
+
+        val cacheUpdateRequired = combo.hashingCompletedSuccessfully && mode.writeToCache
+        val nextStepName = if (cacheUpdateRequired) "Update call cache" else "Update job store"
+        s"$nextStepName when call caching is $mode, the EJEA has $hashComboName and then gets a success result" in {
+          ejea = ejeaInBackendIsCopyingCachedOutputsState(initialHashData, mode)
+          // Send the response from the EJHA (if there was one!):
+          ejhaResponse foreach { ejea ! _ }
+
+          // Nothing should happen here:
+          helper.jobStoreProbe.expectNoMsg(awaitAlmostNothing)
+          helper.callCacheWriteActorCreations.size should be(0) // Rely on the await timeout from the previous step to allow time to pass
+
+          // Send the response from the copying actor
+          ejea ! successResponse
+
+          if (cacheUpdateRequired) {
+            expectCacheWrite(successResponse, finalHashData.get.get)
+          } else {
+            expectJobStoreWrite(SucceededResponseData(successResponse, finalHashData))
+          }
+          // A separate check of the final effective call caching mode:
+          if (ejhaResponse == hashErrorEjhaResponse || initialHashData == hashErrorDataValue) {
+            ejea.underlyingActor.checkEffectiveCallCachingMode should be(mode.withoutWrite)
+          } else {
+            ejea.underlyingActor.checkEffectiveCallCachingMode should be(mode)
+          }
+        }
+
+        s"$nextStepName when it gets a success result and it then gets $hashComboName, if call caching is $mode" in {
+          ejea = ejeaInBackendIsCopyingCachedOutputsState(initialHashData, mode)
+          // Send the response from the copying actor
+          ejea ! successResponse
+
+          ejhaResponse foreach { resp =>
+            // Nothing should have happened yet:
+            helper.jobStoreProbe.expectNoMsg(awaitAlmostNothing)
+            helper.callCacheWriteActorCreations.size should be(0) // Rely on the await timeout from the previous step to allow time to pass
+
+            // Ok, now send the response from the EJHA (if there was one!):
+            ejea ! resp
+          }
+
+          if (cacheUpdateRequired) {
+            expectCacheWrite(successResponse, finalHashData.get.get)
+          } else {
+            expectJobStoreWrite(SucceededResponseData(successResponse, finalHashData))
+          }
+          // A separate check of the final effective call caching mode:
+          if (ejhaResponse == hashErrorEjhaResponse || initialHashData == hashErrorDataValue) {
+            ejea.underlyingActor.checkEffectiveCallCachingMode should be(mode.withoutWrite)
```

Comment:
Did you say this after you saw my response? If you're that sure, I'll change it...